R/utils_data_handling.R

```diff
@@ -1,129 +1,126 @@
+    # do a full, where all data is kept.
+    combined <- merge(observations, forecasts, by = by, all = TRUE)
+  } else {
+    combined <- merge(observations, forecasts, by = by, all.y = TRUE)
+  }
 
-#' @title Pivot Range Format Forecasts From Long to Wide Format
-#'
-#' @description
-#' Legacy function that will not be supported in future updates.
-#' @inheritParams range_long_to_wide
-#' @return a data.frame in wide format
-#' @importFrom data.table dcast
-#' @export
-quantile_to_wide <- function(data) {
-  warning("This function will be deprecated. Please use `range_long_to_wide()` in the future")
-  out <- scoringutils::range_long_to_wide(data)
-  return(out[])
-}
 
+  # get colnames that are the same for x and y
+  colnames <- colnames(combined)
+  colnames_x <- colnames[endsWith(colnames, ".x")]
+  colnames_y <- colnames[endsWith(colnames, ".y")]
 
+  # extract basenames
+  basenames_x <- sub(".x$", "", colnames_x)
+  basenames_y <- sub(".y$", "", colnames_y)
 
+  # see whether the column name as well as the content is the same
+  overlapping <- (as.list(combined[, ..colnames_x]) %in% as.list(combined[, ..colnames_y])) & basenames_x == basenames_y
+  overlap_names <- colnames_x[overlapping]
+  basenames_overlap <- sub(".x$", "", overlap_names)
 
-#' @title Pivot Range Format Forecasts From Wide to Long Format
+  # delete overlapping columns
+  if (length(basenames_overlap > 0)) {
+    combined[, paste0(basenames_overlap, ".x") := NULL]
+    combined[, paste0(basenames_overlap, ".y") := NULL]
+  }
+
+  return(combined[])
+}
+
+#' @title Change Data from a Sample Based Format to a Quantile Format
 #'
 #' @description
-#' Given a data.frame that follows the structure shown in
-#' [range_example_data_wide()], the function outputs the same
-#' data in a long format as (as shown in
-#' [range_example_data_long()]). This can be useful e.g. for
-#' plotting.
 #'
-#' @param data a data.frame following the specifications from
-#' [eval_forecasts()]) for quantile forecasts. For an example, see
-#' [range_example_data_wide()])
-#' @return a data.frame in long format
-#' @importFrom data.table melt
+#' Transform data from a format that is based on predictive samples to a format
+#' based on plain quantiles.
+#'
+#'
+#' @param data a data.frame with samples
+#' @param quantiles a numeric vector of quantiles to extract
+#' @param type type argument passed down to the quantile function. For more
+#' information, see [quantile()]
+#' @return a data.frame in a long interval range format
+#' @importFrom data.table as.data.table
+#' @importFrom stats quantile
+#' @keywords data-handling
 #' @export
 #' @examples
-#' wide <- scoringutils::range_example_data_wide
-#' long <- scoringutils::range_wide_to_long(wide)
-#'
-
-range_wide_to_long <- function(data) {
-
+#' sample_to_quantile(example_integer)
+sample_to_quantile <- function(data,
+                               quantiles = c(0.05, 0.25, 0.5, 0.75, 0.95),
+                               type = 7) {
```

Comment:
Would it be better to use `...` to allow the user to pass any additional argument to `quantile()` instead of just `type`?

Reply:
and then have something like `if (missing(type)) {type = 7}`?
Probably more elegant, we would just have to document it. :+1:

Reply:
Looking at the docs from `quantile()`, `type = 7` already seems to be the default. So you shouldn't have to specify it again if missing.

Reply:
but na.rm is FALSE by default and we want it to be TRUE here probably (hm - or do we?)

Reply:
I made it the default for now